pkg/task/task.go

```diff
@@ -2,42 +2,48 @@ package task
 
 import (
 	"errors"
+	"github.com/Sirupsen/logrus"
 	"time"
 )
 
+//TODO: export the type: type Task func() (string, error)
+
 // ErrTimedOut is returned when an operation times out
 var ErrTimedOut = errors.New("timed out performing task")
 
 // DoRetryWithTimeout performs given task with given timeout and timeBeforeRetry
-func DoRetryWithTimeout(t func() error, timeout, timeBeforeRetry time.Duration) error {
+func DoRetryWithTimeout(t func() (string, error), timeout, timeBeforeRetry time.Duration) (string, error) {
```

Comment:
Why do we need `DoRetryWithTimeout(...)` to return a string output? Unless we have a requirement, let's keep the interface minimal.

Reply:
Change returning type to `interface{}` instead of a string.